File: Data Structures & Algorithms/remove-element/submission-0.cpp
class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        int ans = 0;
        int n = nums.size();
        int i = 0, j = 0;
        while(j<n){
            if(nums[j]!=val){
                nums[i++] = nums[j];
                ans++;
            }
            j++;
        }
        return ans;
    }
};

File: Data Structures & Algorithms/string-encode-and-decode/submission-0.java
class Solution {

    public String encode(List<String> strs) {
        StringBuilder ans = new StringBuilder();
        for(String str: strs){
            int len = str.length();
            ans.append(Integer.toString(len));
            ans.append('#');
            ans.append(str);
        }
        System.out.println(ans.toString());
        return ans.toString();
    }

    public List<String> decode(String str) {
        List<String> ans = new ArrayList<>();
        int i = 0;
        while(i< str.length()){
            StringBuilder st = new StringBuilder();     
            while(str.charAt(i)!='#'){
                st.append(str.charAt(i++));
            }
            int stLen = Integer.parseInt(st.toString());
            StringBuilder tAns = new StringBuilder(stLen);
            int j = ++i;
            for(; j<i+stLen; j++){
                tAns.append(str.charAt(j));
            }
            i = j;
            ans.add(tAns.toString());
        }
        return ans;
    }
}
